source/common/config/delta_subscription_state.cc

```diff
@@ -124,6 +134,29 @@ void DeltaSubscriptionState::handleEstablishmentFailure() {
 
 envoy::api::v2::DeltaDiscoveryRequest DeltaSubscriptionState::getNextRequestAckless() {
   envoy::api::v2::DeltaDiscoveryRequest request;
+  if (!any_request_sent_yet_in_current_stream_) {
```

Comment:
While the simple if conditions are nice, I think it might be easier to follow as a single if and single else.

Reply:
Yeah, realised I could handle aliases in DeltaSubscriptionState::populateDiscoveryRequest, no reason to treat them as a special case. Fixed.